### PR TITLE
✨ 227 - Add app history download endpoint

### PR DESCRIPTION
**Pull request by anncatton**
Adds endpoint for admin tsv download
Simplest approach for now: retrieves all app updates and then sorts by date
Adds endpoint to swagger docs

### CODE REVIEW

File: src/routes/applications.ts [hunk 1]
```diff
@@ -287,6 +285,18 @@ const createApplicationsRouter = (
     }),
   );
 
+  router.get(
+    '/export/application-history/',
+    authFilter([config.auth.REVIEW_SCOPE]),
+    wrapAsync(async (req: Request, res: Response) => {
+      const apps = await getApplicationUpdates();
```
Comment (blabadi) [inline src/routes/applications.ts:292]: I think

File: src/utils/misc.ts [hunk 2]
```diff
@@ -153,3 +154,66 @@ export const getUpdateAuthor: (id: string, isReviewer: boolean) => UpdateAuthor
...
+  { name: 'Applicant', accessor: 'applicant' },
+  { name: 'Project Title', accessor: 'projectTitle' },
+  { name: 'Ethics Letter', accessor: 'ethicsLetterRequired' },
+];
+
+const sortByDate = (a: any, b: any) => {
+  return b.date.getTime() - a.date.getTime();
+};
+
+export const createAppHistoryTSV = (results: ApplicationDocument[]) => {
```
Comment (blabadi) [inline src/utils/misc.ts:180]: this function can be in the service, it contains logic to transform data, and would only be used in the service, my suggestion is to move it there

File: src/routes/applications.ts [hunk 3]
```diff
@@ -287,6 +286,18 @@ const createApplicationsRouter = (
...
+      const tsv = createAppHistoryTSV(apps);
```
Comment (blabadi) [inline src/routes/applications.ts:294]: can't these be one call ? It's always a good practice to keep the router as thin as possible and keep everything pushed in the service

File: src/utils/misc.ts [hunk 4]
```diff
@@ -153,3 +154,25 @@ export const getUpdateAuthor: (id: string, isReviewer: boolean) => UpdateAuthor
...
+  { name: 'Application Status', accessor: 'eventType' },
+  { name: 'Application Type', accessor: 'appType' },
+  { name: 'Action Performed By', accessor: 'role' },
+  { name: 'Days Since Last Status Change', accessor: 'daysElapsed' },
+  { name: 'Institution', accessor: 'institution' },
+  { name: 'Country', accessor: 'country' },
+  { name: 'Applicant', accessor: 'applicant' },
+  { name: 'Project Title', accessor: 'projectTitle' },
+  { name: 'Ethics Letter', accessor: 'ethicsLetterRequired' },
+];
```
Comment (blabadi) [inline src/utils/misc.ts:174]: this could be wrapped in the service file as well, is there a reason I may be missing to have it located here ?

Comment (anncatton) [reply src/utils/misc.ts:174]: nope - actually now that i think about it i might just put this in the tsv function itself, it doesn't need to be accessed by anything else

File: src/routes/applications.ts (hunk 3 above)
Comment (anncatton) [reply src/routes/applications.ts:294]: will do!

Comment (anncatton) [reply src/routes/applications.ts:294]: i even did that for the csv call, not sure why i did it differently here 🤔

**Review by blabadi — APPROVED** [review]

**Pull request merged**